### PR TITLE
Implement Streaming API client for python 3

### PR DESCRIPTION
**Pull request by adamhicks**

### CODE REVIEW

File: examples/stream.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+def handle_update(pair, state, update):
+    print(format_state(pair, state))
+    if update is not None:
+        print(update)
+
+asyncio.get_event_loop().run_until_complete(
+    stream_market(
+        pair="XBTZAR",
+        api_key_id=[api key],
```
Comment (neilgarb) [inline examples/stream.py]: Suggest using a dummy string literal so this at least runs.

File: examples/stream.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        api_key_secret=[api secret],
```
Comment (neilgarb) [inline examples/stream.py]: Same here.

File: luno_python/api_types.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from collections import namedtuple
+from decimal import Decimal
+from typing import List
+
+DEC_0 = Decimal('0')
```
Comment (neilgarb) [inline luno_python/api_types.py]: Inconsistent quotes.

File: luno_python/stream_client.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+def handle_update(pair, state, update):
+    print(format_state(pair, state))
+    if update is not None:
+        print(update)
+
+asyncio.get_event_loop().run_until_complete(
+    stream_market(
+        pair="XBTZAR",
+        api_key_id=[api key],
```
Comment (neilgarb) [inline luno_python/stream_client.py]: See comments above.

File: examples/stream.py (hunk 1 above)
Comment (adamhicks) [reply examples/stream.py]: Add empty string

File: luno_python/api_types.py (hunk 3 above)
Comment (adamhicks) [reply luno_python/api_types.py]: Fixed up quotes.

**Review by neilgarb — DISMISSED** [review]

**Comment by YinYin-blip** [comment]
There might be a critical bug in the streamingapi of this merge. Please check out the pull request before merging: https://github.com/adamhicks/luno-python/pull/1

File: luno_python/stream_client.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+                Decimal(msg['price']),
+                Decimal(msg['volume']),
+            )
+
+        bids = [conv_message(m) for m in first['bids']]
+        asks = [conv_message(m) for m in first['asks']]
+        self._bids = {b.order_id: b for b in bids}
+        self._asks = {a.order_id: a for a in asks}
+        self._sequence = first['sequence']
+        self._trades = []
```
Comment (YinYin-blip) [inline luno_python/stream_client.py:55]: Is this variable used anywhere?

File: luno_python/stream_client.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            Decimal(create_update['volume']),
+        )
+        if create_update['type'] == "ASK":
+            self._asks[o.order_id] = o
+        elif create_update['type'] == "BID":
+            self._bids[o.order_id] = o
+
+    def _process_delete(self, delete_update: dict):
+        order_id = delete_update['order_id']
+        self._asks.pop(order_id, None)
```
Comment (YinYin-blip) [inline luno_python/stream_client.py:125]: Why not `self._asks.pop(order_id)` ? Raising a `KeyError` if `order_id` is not found might be preferred.

File: luno_python/stream_client.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        self._bids.pop(order_id, None)
```
Comment (YinYin-blip) [inline luno_python/stream_client.py:126]: Same as above

**Review by YinYin-blip — COMMENTED** [review]
Unresolved comments.

**Pull request closed without merging**